cloud/linode/route_controller.go

```diff
@@ -0,0 +1,237 @@
+			timeout = t
+		}
+	}
+	klog.V(3).Infof("TTL for routeCache set to %d", timeout)
+
+	vpcid, err := getVPCID(client, Options.VPCName)
+	if err != nil {
+		return nil, err
+	}
+
+	return &routes{
+		vpcid:     vpcid,
+		client:    client,
+		instances: newInstances(client),
+		routeCache: &routeCache{
+			routes: make(map[int][]linodego.InstanceConfig),
+			ttl:    time.Duration(timeout) * time.Second,
+		},
+	}, nil
+}
+
+// instanceConfigsByID returns InstanceConfigs for given instance id
+func (r *routes) instanceConfigsByID(id int) ([]linodego.InstanceConfig, error) {
+	r.routeCache.RLock()
+	defer r.routeCache.RUnlock()
+	instanceConfigs, ok := r.routeCache.routes[id]
+	if !ok {
+		return nil, cloudprovider.InstanceNotFound
+	}
+	return instanceConfigs, nil
+}
+
+// getInstanceConfigs returns InstanceConfigs for given instance id
+// It refreshes routeCache if it has expired
+func (r *routes) getInstanceConfigs(ctx context.Context, id int) ([]linodego.InstanceConfig, error) {
+	if err := r.routeCache.refreshRoutes(ctx, r.client); err != nil {
+		return nil, err
+	}
+
+	return r.instanceConfigsByID(id)
+}
+
+// getInstanceFromName returns linode instance with given name if it exists
+func (r *routes) getInstanceFromName(ctx context.Context, name string) (*linodego.Instance, error) {
+	// create node object
+	node := &v1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+	}
+
+	// fetch instance with specified node name
+	instance, err := r.instances.lookupLinode(ctx, node)
+	if err != nil {
+		klog.Errorf("failed getting linode %s", name)
+		return nil, err
+	}
+	return instance, nil
+}
+
+// CreateRoute adds route's subnet to ip_ranges of target node's VPC interface
+func (r *routes) CreateRoute(ctx context.Context, clusterName string, nameHint string, route *cloudprovider.Route) error {
+	instance, err := r.getInstanceFromName(ctx, string(route.TargetNode))
+	if err != nil {
+		return err
+	}
+
+	// fetch instance configs
+	configs, err := r.getInstanceConfigs(ctx, instance.ID)
+	if err != nil {
+		return err
+	}
+
+	// find VPC interface and add route to it
+	for _, iface := range configs[0].Interfaces {
+		if iface.VPCID != nil && r.vpcid == *iface.VPCID && iface.IPv4.VPC != "" {
+			if slices.Contains(iface.IPRanges, route.DestinationCIDR) {
+				klog.V(4).Infof("Route already exists for node %s", route.TargetNode)
+				return nil
+			}
+
+			ipRanges := append(iface.IPRanges, route.DestinationCIDR)
+			interfaceUpdateOptions := linodego.InstanceConfigInterfaceUpdateOptions{
+				IPRanges: ipRanges,
+			}
+			resp, err := r.client.UpdateInstanceConfigInterface(ctx, instance.ID, configs[0].ID, iface.ID, interfaceUpdateOptions)
+			if err != nil {
+				return err
+			}
+			klog.V(4).Infof("Added routes for node %s. Current routes: %v", route.TargetNode, resp.IPRanges)
+			return nil
+		}
+	}
+
+	klog.V(4).Infof("Unable to add route for node %s. No valid interface found", route.TargetNode)
+	return nil
```

Comment:
It's not clear to me why this is not an error, the same thing in DeleteRoute

Reply:
Changed it from info to error at both places